privilege-request-protocol.md

```diff
@@ -0,0 +1,208 @@
+   privilege, so includes in the response a `Link` for the privilege request
+   endpoint URI.
+3. The client determines to request additional privilege and makes a simple
+   `POST` to the provided privilege request endpoint URI without credentials.
+4. The privilege request endpoint determines the request is acceptable and
+   queues the request, returning an HTTP `202 Accepted` response.
+
+The privilege management system might (but is not required to) provide a
+web-based permission management interface, and in some circumstances it might
+be appropriate to allow the client's user to access this interface. The
+privilege request endpoint **MAY** include in its response a URI [template][RFC6570]
+to access this interface, to which the client can redirect the user's trusted
+web browser. The interface URI template **SHOULD** include a `redirect_uri`
+variable so that the interface can eventually redirect back to the client
+application on completion of the management activity.
+
+On completion of the management activity (for example, but not limited to,
+in a permission management interface if provided, or offline by whatever means
+supported by the server) the client can attempt access again. There is no
+guarantee that sufficient privilege has been granted at this point, and new
+requests might still be denied, potentially restarting this process.
+
+Syntax
+------
+The link to the privilege request endpoint is in an HTTP `Link` response
+header of the original denied transaction. The link relation is *TBD*, but for
+now let us use `x-permission-request` as a placeholder pending an agreeable
+identifier. The link target URI is opaque to the client, need not be for the
+same origin as the original denied request, **SHOULD** be unguessable and
+unforgeable, and **SHOULD** only be valid for a limited time and a single
+use.  The privilege request endpoint **MUST** be able to sufficiently identify
+the client solely by this URI, for example by the URI comprising a key to a
+database record, or by comprising a signed serialization of relevant information
+about the client.  The `Link` header **SHOULD** include an `expires_in`
+parameter giving the number of seconds after the `Date` of this response at
+which the privilege request URI will no longer be valid.
+
+A request for additional privilege **SHALL** be made by HTTP `POST` to the
+privilege request URI. The endpoint **MUST NOT** require any special credentials,
+including any authentication or authorization credentials included with the
+original request; that is, the URI itself is a sufficient [capability][]
+for this specific request.
+
+The request body of the `POST` to the privilege request endpoint, if present,
+**SHALL** have a content-type of `application/x-www-form-urlencoded`. At this
+time no parameters are defined. A possible future extension could include a
+[webhook][] to be notified on completion of the management activity. The
```

Comment:
todo: add link?

Reply:
i'm not sure what you're asking. if you mean for the "empty" `[webhook][]` link, it's at the bottom of the document.

Reply:
ah ok